frontend/services/template.service.ts

```diff
@@ -0,0 +1,41 @@
+import { useFetch } from '#app';
+import type { TemplateSchema, FormData } from '@/types/template';
+
+/**
+ * Service for handling template-related API operations
+ */
+class TemplateService {
```

Comment:
Use a [composable ](https://vuejs.org/guide/reusability/composables) instead of a class for reusable logic